[CI] GenerativeRandomMappingIT test failing

**Build Scans:**
- [elasticsearch-intake #38487 / part3](https://gradle-enterprise.elastic.co/s/bkv4dudf3oyds)
- [elasticsearch-pull-request #140776 / part-3](https://gradle-enterprise.elastic.co/s/ko647ssa7kzl2)
- [elasticsearch-pull-request #140724 / part-3](https://gradle-enterprise.elastic.co/s/uvwfi4dmsiiya)
- [elasticsearch-pull-request #140724 / part-3](https://gradle-enterprise.elastic.co/s/6xxx7w2efqmnm)
- [elasticsearch-pull-request #140724 / part-3](https://gradle-enterprise.elastic.co/s/w2ipni76dvpr4)
- [elasticsearch-pull-request #140635 / part-3](https://gradle-enterprise.elastic.co/s/y7yicxz6ampcu)
- [elasticsearch-pull-request #140639 / part-3](https://gradle-enterprise.elastic.co/s/24r4m63a2rdmk)
- [elasticsearch-pull-request #140635 / part-3](https://gradle-enterprise.elastic.co/s/u7bho2iqfnn6c)
- [elasticsearch-pull-request #140639 / part-3](https://gradle-enterprise.elastic.co/s/r6unopyznw6pq)
- [elasticsearch-pull-request #140635 / part-3](https://gradle-enterprise.elastic.co/s/rh2obdhil3clg)

**Reproduction Line:**
```
./gradlew ":x-pack:plugin:esql:qa:server:single-node:javaRestTest" --tests "org.elasticsearch.xpack.esql.qa.single_node.GenerativeRandomMappingIT.test" -Dtests.seed=FE1262A9DA8A7581 -Dtests.locale=en-IO -Dtests.timezone=Africa/Bissau -Druntime.java=26
```

**Applicable branches:**
main

**Reproduces locally?:**
N/A

**Failure History:**
[See dashboard](https://es-delivery-stats.elastic.dev/app/dashboards#/view/dcec9e60-72ac-11ee-8f39-55975ded9e63?_g=(refreshInterval:(pause:!t,value:60000),time:(from:now-7d%2Fd,to:now))&_a=(controlGroupState:(initialChildControlState:('0c0c9cb8-ccd2-45c6-9b13-96bac4abc542':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:task.keyword,order:0,selectedOptions:!(),title:'GradleTask',type:optionsListControl),'4e6ad9d6-6fdc-4fcc-bf1a-aa6ca79e0850':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:className.keyword,order:1,selectedOptions:!(org.elasticsearch.xpack.esql.qa.single_node.GenerativeRandomMappingIT),title:'Suite',type:optionsListControl),'144933da-5c1b-4257-a969-7f43455a7901':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:name.keyword,order:2,selectedOptions:!('test'),title:'Test',type:optionsListControl)))))

**Failure Message:**
```
java.lang.AssertionError: query: SET approximation={"confidence_level":0.6964470704293608};from gen_rm_*,gen_rm_*,ge* | SAMPLE 0.6345448750192555 | keep mode, `height`, category, `category` | FORK  (where true) (where  NOT length(category) > 7 AND  NOT true) (where  NOT contains(category, "Xz") OR  NOT match(mode, "fox") | LIMIT 99 BY category) (where match_phrase(category, "fox brown") OR byte_length(category) < 12) (where mode IN (22, 85, 13) AND  NOT false) (where  NOT starts_with(category, "W")) (where  NOT length(category) >= 12) (where  NOT false) | WHERE _fork == "fork4" | DROP _fork
exception: method [POST], host [http://[::1]:36557], URI [/_query?pretty=true&error_trace=true], status line [HTTP/1.1 400 Bad Request]
Warnings: [No limit defined, adding default limit of [1000], line 1:59: approximation not supported: query must have [STATS] with aggregation function(s) that can be approximated]
---
error:
  root_cause:
  - type: "query_shard_exception"
    reason: "failed to crea
[truncated]
```

**Issue Reasons:**
- [main] 1 consecutive failures in test test
- [main] 1 consecutive failures in step part3
- [main] 1 consecutive failures in pipeline elasticsearch-intake
- [main] 29 failures in test test (2.9% fail rate in 988 executions)
- [main] 26 failures in step part-3 (4.2% fail rate in 618 executions)
- [main] 2 failures in step part-3-fips-140-3 (7.7% fail rate in 26 executions)
- [main] 15 failures in pipeline elasticsearch-pull-request (2.5% fail rate in 602 executions)

**Note:**
This issue was created using new test triage automation. Please report issues or feedback to es-delivery.

--- a/muted-tests.yml
+++ b/muted-tests.yml
@@ -333,9 +333,6 @@ tests:
 - class: org.elasticsearch.multiproject.test.CoreWithMultipleProjectsClientYamlTestSuiteIT
   method: test {yaml=reindex/10_basic/Response format for version conflict with conflicts=proceed}
   issue: https://github.com/elastic/elasticsearch/issues/147594
-- class: org.elasticsearch.xpack.esql.qa.single_node.GenerativeRandomMappingIT
-  method: test
-  issue: https://github.com/elastic/elasticsearch/issues/147610
 - class: org.elasticsearch.xpack.esql.qa.single_node.EsqlSpecForceStoredLoadingIT
   method: test {csv-spec:fuse.fuseWithRowLinearAndMultiValueGroupColumn}
   issue: https://github.com/elastic/elasticsearch/issues/147613

--- a/x-pack/plugin/esql/qa/server/src/main/java/org/elasticsearch/xpack/esql/qa/rest/generative/GenerativeRandomMappingRestTest.java
+++ b/x-pack/plugin/esql/qa/server/src/main/java/org/elasticsearch/xpack/esql/qa/rest/generative/GenerativeRandomMappingRestTest.java
@@ -71,6 +71,14 @@ public abstract class GenerativeRandomMappingRestTest extends GenerativeRestTest
         Pattern.DOTALL
     );
 
+    // Full-text functions (match, match_phrase, qstr) pushed down to Lucene can fail with
+    // "failed to create query" on random mappings where field types are incompatible (e.g., non-indexed
+    // fields, numeric fields receiving text input). https://github.com/elastic/elasticsearch/issues/147610
+    private static final Pattern FAILED_TO_CREATE_QUERY = Pattern.compile(
+        ".*failed to create query.*",
+        Pattern.DOTALL
+    );
+
     @Before
     public void setupRandomIndices() throws IOException {
         synchronized (GenerativeRandomMappingRestTest.class) {
@@ -232,6 +240,12 @@ public abstract class GenerativeRandomMappingRestTest extends GenerativeRestTest
         if (query != null && isAllowedError(errorMessage, UNSUPPORTED_TYPE_IN_FUNCTION) && queryContainsFork(query)) {
             return true;
         }
+        // Full-text functions pushed to Lucene can throw QueryShardException("failed to create query: ...")
+        // when the target field has an incompatible type or is not indexed in the random mapping.
+        // https://github.com/elastic/elasticsearch/issues/147610
+        if (query != null && isAllowedError(errorMessage, FAILED_TO_CREATE_QUERY) && queryContainsFullTextFunction(query)) {
+            return true;
+        }
         return false;
     }
 
@@ -252,6 +266,14 @@ public abstract class GenerativeRandomMappingRestTest extends GenerativeRestTest
 
     private static boolean queryContainsFork(String query) {
         return query != null && query.toUpperCase(Locale.ROOT).contains("FORK");
+    }
+
+    private static boolean queryContainsFullTextFunction(String query) {
+        if (query == null) {
+            return false;
+        }
+        String upper = query.toUpperCase(Locale.ROOT);
+        return upper.contains("MATCH(") || upper.contains("MATCH_PHRASE(") || upper.contains("QSTR(") || upper.contains("KQL(");
     }
 
     private static String formatReproductionContext(String query) {

--- a/x-pack/plugin/esql/qa/server/src/main/java/org/elasticsearch/xpack/esql/qa/rest/generative/GenerativeRandomMappingRestTest.java
+++ b/x-pack/plugin/esql/qa/server/src/main/java/org/elasticsearch/xpack/esql/qa/rest/generative/GenerativeRandomMappingRestTest.java
@@ -74,10 +74,7 @@ public abstract class GenerativeRandomMappingRestTest extends GenerativeRestTest
     // Full-text functions (match, match_phrase, qstr) pushed down to Lucene can fail with
     // "failed to create query" on random mappings where field types are incompatible (e.g., non-indexed
     // fields, numeric fields receiving text input). https://github.com/elastic/elasticsearch/issues/147610
-    private static final Pattern FAILED_TO_CREATE_QUERY = Pattern.compile(
-        ".*failed to create query.*",
-        Pattern.DOTALL
-    );
+    private static final Pattern FAILED_TO_CREATE_QUERY = Pattern.compile(".*failed to create query.*", Pattern.DOTALL);
 
     @Before
     public void setupRandomIndices() throws IOException {

